The job **`i686-msvc-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/28132398764/job/83311666930) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/83311666930) [(plain)](https://github.com/rust-lang/rust/commit/3bfa380d5a77e450d2d125816de21e242fea4b32/checks/83311666930/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[endgroup]
[TIMING:end] tool::ToolBuild { build_compiler: Compiler { stage: 0, host: i686-pc-windows-msvc, forced_compiler: false }, target: i686-pc-windows-msvc, tool: "linkchecker", path: "src/tools/linkchecker", mode: ToolBootstrap, source_type: InTree, extra_features: [], allow_features: "", cargo_args: [], artifact_kind: Binary } -- 1.187
[TIMING:end] tool::Linkchecker { compiler: Compiler { stage: 0, host: i686-pc-windows-msvc, forced_compiler: false }, target: i686-pc-windows-msvc } -- 0.000
##[group]Testing stage1 Linkcheck (i686-pc-windows-msvc)
cargo\commands\cargo-bench.html:483: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-build.html:404: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-check.html:400: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-clean.html:276: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-doc.html:374: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-fetch.html:219: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-fix.html:485: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-install.html:450: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-package.html:432: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-publish.html:329: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-run.html:289: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-rustc.html:382: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-rustdoc.html:389: broken link - `rustc\targets\custom.html`
cargo\commands\cargo-test.html:502: broken link - `rustc\targets\custom.html`
cargo\print.html:1880: broken link - `rustc\lints\levels.html`
cargo\print.html:2023: broken link - `rustc\tests\index.html`
cargo\print.html:2147: broken link - `rustc\command-line-arguments.html`
cargo\print.html:4153: broken link - `rustc\command-line-arguments.html`
cargo\print.html:4634: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4652: broken link - `rustc\profile-guided-optimization.html`
cargo\print.html:4655: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4665: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4672: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4676: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4678: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4685: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4699: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4711: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4720: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4720: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4721: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4736: broken link - `rustc\linker-plugin-lto.html`
cargo\print.html:4740: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4756: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4771: broken link - `rustc\codegen-options\index.html`
cargo\print.html:4779: broken link - `rustc\codegen-options\index.html`
cargo\print.html:5447: broken link - `rustc\targets\custom.html`
cargo\print.html:6181: broken link - `rustc\codegen-options\index.html`
cargo\print.html:6812: broken link - `rustc\codegen-options\index.html`
cargo\print.html:6818: broken link - `rustc\codegen-options\index.html`
cargo\print.html:6824: broken link - `rustc\codegen-options\index.html`
cargo\print.html:6828: broken link - `rustc\codegen-options\index.html`
cargo\print.html:6832: broken link - `rustc\codegen-options\index.html`
cargo\print.html:6835: broken link - `rustc\codegen-options\index.html`
cargo\print.html:6838: broken link - `rustc\codegen-options\index.html`
cargo\print.html:6842: broken link - `rustc\command-line-arguments.html`
cargo\print.html:6854: broken link - `rustc\command-line-arguments.html`
cargo\print.html:6856: broken link - `rustc\command-line-arguments.html`
cargo\print.html:6860: broken link - `rustc\command-line-arguments.html`
cargo\print.html:6874: broken link - `rustc\command-line-arguments.html`
cargo\print.html:6877: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\print.html:6890: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\print.html:6891: broken link - `rustc\command-line-arguments.html`
cargo\print.html:6892: broken link - `rustc\check-cfg.html`
cargo\print.html:6905: broken link - `rustc\command-line-arguments.html`
cargo\print.html:7796: broken link - `rustc\json.html`
cargo\print.html:10705: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\print.html:10917: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\print.html:10918: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\print.html:10919: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\print.html:10922: broken link - `rustc\lints\levels.html`
cargo\print.html:11871: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\print.html:12982: broken link - `rustc\codegen-options\index.html`
cargo\print.html:14447: broken link - `rustc\targets\custom.html`
cargo\print.html:14968: broken link - `rustc\targets\custom.html`
cargo\print.html:15516: broken link - `rustc\targets\custom.html`
cargo\print.html:15934: broken link - `rustc\targets\custom.html`
cargo\print.html:16330: broken link - `rustc\targets\custom.html`
cargo\print.html:16671: broken link - `rustc\targets\custom.html`
cargo\print.html:17157: broken link - `rustc\targets\custom.html`
cargo\print.html:17612: broken link - `rustc\targets\custom.html`
cargo\print.html:18120: broken link - `rustc\targets\custom.html`
cargo\print.html:18684: broken link - `rustc\targets\custom.html`
cargo\print.html:19324: broken link - `rustc\targets\custom.html`
cargo\print.html:23281: broken link - `rustc\targets\custom.html`
cargo\print.html:24895: broken link - `rustc\targets\custom.html`
cargo\print.html:25303: broken link - `rustc\targets\custom.html`
cargo\reference\build-scripts.html:334: broken link - `rustc\codegen-options\index.html`
cargo\reference\build-scripts.html:340: broken link - `rustc\codegen-options\index.html`
cargo\reference\build-scripts.html:346: broken link - `rustc\codegen-options\index.html`
cargo\reference\build-scripts.html:350: broken link - `rustc\codegen-options\index.html`
cargo\reference\build-scripts.html:354: broken link - `rustc\codegen-options\index.html`
cargo\reference\build-scripts.html:357: broken link - `rustc\codegen-options\index.html`
cargo\reference\build-scripts.html:360: broken link - `rustc\codegen-options\index.html`
cargo\reference\build-scripts.html:364: broken link - `rustc\command-line-arguments.html`
cargo\reference\build-scripts.html:376: broken link - `rustc\command-line-arguments.html`
cargo\reference\build-scripts.html:378: broken link - `rustc\command-line-arguments.html`
cargo\reference\build-scripts.html:382: broken link - `rustc\command-line-arguments.html`
cargo\reference\build-scripts.html:396: broken link - `rustc\command-line-arguments.html`
cargo\reference\build-scripts.html:399: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\reference\build-scripts.html:412: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\reference\build-scripts.html:413: broken link - `rustc\command-line-arguments.html`
cargo\reference\build-scripts.html:414: broken link - `rustc\check-cfg.html`
cargo\reference\build-scripts.html:427: broken link - `rustc\command-line-arguments.html`
cargo\reference\cargo-targets.html:262: broken link - `rustc\tests\index.html`
cargo\reference\cargo-targets.html:386: broken link - `rustc\command-line-arguments.html`
cargo\reference\config.html:682: broken link - `rustc\targets\custom.html`
cargo\reference\config.html:1416: broken link - `rustc\codegen-options\index.html`
cargo\reference\external-tools.html:251: broken link - `rustc\json.html`
cargo\reference\features.html:222: broken link - `rustc\command-line-arguments.html`
cargo\reference\manifest.html:641: broken link - `rustc\lints\levels.html`
cargo\reference\profiles.html:211: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:229: broken link - `rustc\profile-guided-optimization.html`
cargo\reference\profiles.html:232: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:242: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:249: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:253: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:255: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:262: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:276: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:288: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:297: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:297: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:298: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:313: broken link - `rustc\linker-plugin-lto.html`
cargo\reference\profiles.html:317: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:333: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:348: broken link - `rustc\codegen-options\index.html`
cargo\reference\profiles.html:356: broken link - `rustc\codegen-options\index.html`
cargo\reference\semver.html:1793: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\reference\semver.html:2005: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\reference\semver.html:2006: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\reference\semver.html:2007: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\reference\semver.html:2010: broken link - `rustc\lints\levels.html`
cargo\reference\unstable.html:514: broken link - `rustc\lints\listing\warn-by-default.html`
cargo\reference\unstable.html:1625: broken link - `rustc\codegen-options\index.html`
edition-guide\editions\advanced-migrations.html:185: broken link - `rustc\lints\index.html`
edition-guide\editions\advanced-migrations.html:191: broken link - `rustc\lints\groups.html`
edition-guide\editions\advanced-migrations.html:226: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\editions\advanced-migrations.html:229: broken link - `rustc\lints\groups.html`
edition-guide\editions\advanced-migrations.html:277: broken link - `rustc\lints\groups.html`
edition-guide\editions\advanced-migrations.html:278: broken link - `rustc\lints\levels.html`
edition-guide\editions\advanced-migrations.html:287: broken link - `rustc\json.html`
edition-guide\editions\advanced-migrations.html:306: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\editions\advanced-migrations.html:307: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:367: broken link - `rustc\lints\index.html`
edition-guide\print.html:373: broken link - `rustc\lints\groups.html`
edition-guide\print.html:408: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:411: broken link - `rustc\lints\groups.html`
edition-guide\print.html:459: broken link - `rustc\lints\groups.html`
edition-guide\print.html:460: broken link - `rustc\lints\levels.html`
edition-guide\print.html:469: broken link - `rustc\json.html`
edition-guide\print.html:488: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:489: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:1514: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:1540: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2017: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2037: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2142: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2237: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2289: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2329: broken link - `rustc\symbol-mangling\index.html`
edition-guide\print.html:2347: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2367: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2371: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2394: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2408: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\print.html:2412: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\print.html:2667: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\print.html:2668: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\print.html:2679: broken link - `rustc\lints\listing\deny-by-default.html`
edition-guide\print.html:2720: broken link - `rustc\lints\listing\deny-by-default.html`
edition-guide\print.html:2827: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2870: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2930: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:2993: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\print.html:3042: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\print.html:3059: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\print.html:3084: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2021\raw-lifetimes.html:192: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2021\reserved-syntax.html:230: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\gen-keyword.html:192: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\intoiterator-box-slice.html:222: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\rust-2024\intoiterator-box-slice.html:239: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\rust-2024\macro-fragment-specifiers.html:203: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\match-ergonomics.html:264: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\never-type-fallback.html:187: broken link - `rustc\lints\listing\deny-by-default.html`
edition-guide\rust-2024\never-type-fallback.html:228: broken link - `rustc\lints\listing\deny-by-default.html`
edition-guide\rust-2024\newly-unsafe-functions.html:206: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\prelude.html:232: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\reserved-syntax.html:217: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\static-mut-references.html:186: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\rust-2024\static-mut-references.html:190: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\rust-2024\static-mut-references.html:445: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\rust-2024\static-mut-references.html:446: broken link - `rustc\lints\listing\warn-by-default.html`
edition-guide\rust-2024\temporary-if-let-scope.html:231: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\temporary-if-let-scope.html:251: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\temporary-tail-expr-scope.html:242: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\unsafe-attributes.html:206: broken link - `rustc\symbol-mangling\index.html`
edition-guide\rust-2024\unsafe-attributes.html:224: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\unsafe-extern.html:210: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\unsafe-op-in-unsafe-fn.html:186: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\unsafe-op-in-unsafe-fn.html:190: broken link - `rustc\lints\listing\allowed-by-default.html`
edition-guide\rust-2024\unsafe-op-in-unsafe-fn.html:213: broken link - `rustc\lints\listing\allowed-by-default.html`
index.html:119: broken link - `rustc\index.html`
reference\attributes\codegen.html:579: broken link - `rustc\lints\listing\warn-by-default.html`
reference\attributes\codegen.html:1087: broken link - `rustc\codegen-options\index.html`
reference\attributes\codegen.html:1087: broken link - `rustc\codegen-options\index.html`
reference\attributes\debugger.html:517: broken link - `rustc\codegen-options\index.html`
reference\attributes\diagnostics.html:242: broken link - `rustc\lints\index.html`
reference\attributes\diagnostics.html:307: broken link - `rustc\lints\levels.html`
reference\attributes\diagnostics.html:307: broken link - `rustc\lints\levels.html`
reference\conditional-compilation.html:726: broken link - `rustc\command-line-arguments.html`
reference\conditional-compilation.html:1107: broken link - `rustc\command-line-arguments.html`
reference\index.html:209: broken link - `rustc\index.html`
reference\introduction.html:209: broken link - `rustc\index.html`
reference\items\external-blocks.html:922: broken link - `rustc\command-line-arguments.html`
reference\items\external-blocks.html:939: broken link - `rustc\command-line-arguments.html`
reference\items\external-blocks.html:960: broken link - `rustc\command-line-arguments.html`
reference\linkage.html:407: broken link - `rustc\codegen-options\index.html`
reference\linkage.html:415: broken link - `rustc\lints\listing\allowed-by-default.html`
reference\panic.html:282: broken link - `rustc\codegen-options\index.html`
reference\panic.html:303: broken link - `rustc\codegen-options\index.html`
reference\print.html:207: broken link - `rustc\index.html`
reference\print.html:11398: broken link - `rustc\command-line-arguments.html`
reference\print.html:11779: broken link - `rustc\command-line-arguments.html`
reference\print.html:19438: broken link - `rustc\command-line-arguments.html`
reference\print.html:19455: broken link - `rustc\command-line-arguments.html`
reference\print.html:19476: broken link - `rustc\command-line-arguments.html`
reference\print.html:22798: broken link - `rustc\lints\index.html`
reference\print.html:22863: broken link - `rustc\lints\levels.html`
reference\print.html:22863: broken link - `rustc\lints\levels.html`
reference\print.html:24162: broken link - `rustc\lints\listing\warn-by-default.html`
reference\print.html:24670: broken link - `rustc\codegen-options\index.html`
reference\print.html:24670: broken link - `rustc\codegen-options\index.html`
reference\print.html:25530: broken link - `rustc\codegen-options\index.html`
reference\print.html:49772: broken link - `rustc\codegen-options\index.html`
reference\print.html:49793: broken link - `rustc\codegen-options\index.html`
reference\print.html:50102: broken link - `rustc\codegen-options\index.html`
reference\print.html:50110: broken link - `rustc\lints\listing\allowed-by-default.html`
reference\print.html:53364: broken link - `rustc\lints\listing\allowed-by-default.html`
reference\print.html:53393: broken link - `rustc\lints\listing\allowed-by-default.html`
reference\print.html:53405: broken link - `rustc\lints\listing\allowed-by-default.html`
reference\unsafe-keyword.html:200: broken link - `rustc\lints\listing\allowed-by-default.html`
reference\unsafe-keyword.html:229: broken link - `rustc\lints\listing\allowed-by-default.html`
reference\unsafe-keyword.html:241: broken link - `rustc\lints\listing\allowed-by-default.html`
std\attribute.must_use.html:4: broken link - `rustc\lints\listing\warn-by-default.html`
unstable-book\compiler-environment-variables\RUSTC_BOOTSTRAP.html:193: broken link - `rustc\lints\listing\allowed-by-default.html`
unstable-book\compiler-flags\extern-options.html:202: broken link - `rustc\lints\listing\warn-by-default.html`
unstable-book\compiler-flags\extern-options.html:203: broken link - `rustc\lints\listing\allowed-by-default.html`
unstable-book\compiler-flags\virtual-function-elimination.html:186: broken link - `rustc\codegen-options\index.html`
unstable-book\print.html:236: broken link - `rustc\lints\listing\allowed-by-default.html`
unstable-book\print.html:1178: broken link - `rustc\lints\listing\warn-by-default.html`
unstable-book\print.html:1179: broken link - `rustc\lints\listing\allowed-by-default.html`
unstable-book\print.html:3531: broken link - `rustc\codegen-options\index.html`
checked links in: 48.3s
number of HTML files scanned: 62075
number of HTML redirects found: 22171
number of links checked: 2939005
number of links ignored due to external: 181534
number of links ignored due to exceptions: 28
number of intra doc links ignored: 9
errors found: 245
NOTE: if you are adding or renaming a markdown file in a mdBook, don't forget to register the page in SUMMARY.md
found some broken links
Command `D:\a\rust\rust\build\i686-pc-windows-msvc\stage1-tools-bin\linkchecker.exe D:\a\rust\rust\build\i686-pc-windows-msvc\doc` failed with exit code 1
Created at: src\bootstrap\src\core\build_steps\tool.rs:1625:23
Executed at: src\bootstrap\src\core\build_steps\test.rs:203:77

--- BACKTRACE vvv
   0: std::backtrace::Backtrace::capture
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\std\src\backtrace.rs:296
   1: bootstrap::utils::exec::DeferredCommand::finish_process
             at .\src\bootstrap\src\utils\exec.rs:939
   2: bootstrap::utils::exec::DeferredCommand::wait_for_output<ref$<bootstrap::utils::exec::ExecutionContext> >
             at .\src\bootstrap\src\utils\exec.rs:831
   3: bootstrap::utils::exec::ExecutionContext::run
             at .\src\bootstrap\src\utils\exec.rs:741
   4: bootstrap::utils::exec::BootstrapCommand::run<ref$<bootstrap::core::builder::Builder> >
             at .\src\bootstrap\src\utils\exec.rs:339
   5: bootstrap::core::build_steps::test::impl$1::run
             at .\src\bootstrap\src\core\build_steps\test.rs:203
   6: bootstrap::core::builder::Builder::ensure<bootstrap::core::build_steps::test::Linkcheck>
             at .\src\bootstrap\src\core\builder\mod.rs:1597
   7: bootstrap::core::build_steps::test::impl$1::make_run
             at .\src\bootstrap\src\core\build_steps\test.rs:140
   8: bootstrap::core::builder::StepDescription::maybe_run
             at .\src\bootstrap\src\core\builder\mod.rs:476
   9: bootstrap::core::builder::cli_paths::match_paths_to_steps_and_run
             at .\src\bootstrap\src\core\builder\cli_paths.rs:141
  10: bootstrap::core::builder::Builder::run_step_descriptions
             at .\src\bootstrap\src\core\builder\mod.rs:1140
  11: bootstrap::Build::build
             at .\src\bootstrap\src\lib.rs:803
  12: bootstrap::Build::build
             at .\src\bootstrap\src\lib.rs:803
  13: bootstrap::main
             at .\src\bootstrap\src\bin\main.rs:130
  14: core::ops::function::FnOnce::call_once<void (*)(),tuple$<> >
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\core\src\ops\function.rs:250
  15: std::sys::backtrace::__rust_begin_short_backtrace<void (*)(),tuple$<> >
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\std\src\sys\backtrace.rs:166
  16: std::sys::backtrace::__rust_begin_short_backtrace<void (*)(),tuple$<> >
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\std\src\sys\backtrace.rs:166
  17: std::rt::lang_start_internal::closure$0
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\std\src\rt.rs:175
  18: std::panicking::catch_unwind::do_call
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\std\src\panicking.rs:581
  19: std::panicking::catch_unwind
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\std\src\panicking.rs:544
  20: std::panic::catch_unwind
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\std\src\panic.rs:359
  21: std::rt::lang_start_internal
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\std\src\rt.rs:171
  22: std::rt::lang_start<tuple$<> >
             at /rustc/0417c25868d6dfbd1c291dfeae950504faa6f790/library\std\src\rt.rs:205
  23: invoke_main
             at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  24: __scrt_common_main_seh
             at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  25: BaseThreadInitThunk
  26: RtlInitializeExceptionChain
  27: RtlGetAppContainerNamedObjectPath


Command has failed. Rerun with -v to see more details.
Bootstrap failed while executing `test --stage 2 --skip=tests --skip=coverage-map --skip=coverage-run --skip=library --skip=tidyselftest`
Build completed unsuccessfully in 1:58:01
make: *** [Makefile:117: ci-msvc-ps1] Error 1
  local time: Thu Jun 25 00:06:20 CUT 2026
  network time: Thu, 25 Jun 2026 00:06:20 GMT
##[error]Process completed with exit code 2.
##[group]Run echo "disk usage:"
echo "disk usage:"
```

</details>

--- a/src/bootstrap/mk/Makefile.in
+++ b/src/bootstrap/mk/Makefile.in
@@ -112,9 +112,9 @@ TEST_SET2 := --skip=tests --skip=coverage-map --skip=coverage-run --skip=library
 
 # this intentionally doesn't use `$(BOOTSTRAP)` so we can test the shebang on Windows
 ci-msvc-py:
-	$(Q)$(CFG_SRC_DIR)/x.py test --stage 2 $(TEST_SET1)
+	$(Q)$(CFG_SRC_DIR)/x.py test --stage 2 $(TEST_SET1) --skip=src/tools/linkchecker
 ci-msvc-ps1:
-	$(Q)$(CFG_SRC_DIR)/x.ps1 test --stage 2 $(TEST_SET2)
+	$(Q)$(CFG_SRC_DIR)/x.ps1 test --stage 2 $(TEST_SET2) --skip=src/tools/linkchecker
 ci-msvc: ci-msvc-py ci-msvc-ps1
 
 ## MingW native builders

--- a/src/bootstrap/src/core/build_steps/doc.rs
+++ b/src/bootstrap/src/core/build_steps/doc.rs
@@ -1393,6 +1393,7 @@ impl Step for RustcBook {
     /// "rustbook" is used to convert it to HTML.
     fn run(self, builder: &Builder<'_>) {
         // FIXME: Temporary workaround for https://github.com/rust-lang/rust/issues/158378
+        #[cfg(not(test))] // So this check doesn't affect the bootstrap tests
         if self.target == "i686-pc-windows-msvc" {
             eprintln!("WARNING: Skipping rustc book build to work around #158378");
             return;

--- a/src/bootstrap/src/core/build_steps/doc.rs
+++ b/src/bootstrap/src/core/build_steps/doc.rs
@@ -1392,6 +1392,12 @@ impl Step for RustcBook {
     /// in the "md-doc" directory in the build output directory. Then
     /// "rustbook" is used to convert it to HTML.
     fn run(self, builder: &Builder<'_>) {
+        // FIXME: Temporary workaround for https://github.com/rust-lang/rust/issues/158378
+        if self.target == "i686-pc-windows-msvc" {
+            eprintln!("WARNING: Skipping rustc book build to work around #158378");
+            return;
+        }
+
         let out_base = builder.md_doc_out(self.target).join("rustc");
         t!(fs::create_dir_all(&out_base));
         let out_listing = out_base.join("src/lints");

--- a/src/ci/github-actions/jobs.yml
+++ b/src/ci/github-actions/jobs.yml
@@ -161,7 +161,7 @@ pr:
     env:
       CODEGEN_BACKENDS: gcc
     <<: *job-linux-4c
-  
+
   # This job tests features we want to stabilize soon, to ensure they don't
   # regress.
   - name: x86_64-gnu-next-trait-solver-polonius
@@ -720,6 +720,7 @@ auto:
         --target=i686-pc-windows-msvc
         --enable-full-tools
         --enable-profiler
+        --disable-docs
       SCRIPT: python x.py dist bootstrap --include-default-paths
       DIST_REQUIRE_ALL_TOOLS: 1
       CODEGEN_BACKENDS: llvm,cranelift

--- a/src/tools/build-manifest/src/main.rs
+++ b/src/tools/build-manifest/src/main.rs
@@ -26,6 +26,7 @@ static DOCS_FALLBACK: &[(&str, &str)] = &[
     ("-apple-", "aarch64-apple-darwin"),
     ("aarch64", "aarch64-unknown-linux-gnu"),
     ("arm-", "aarch64-unknown-linux-gnu"),
+    ("i686-pc-windows", "x86_64-pc-windows-msvc"),
     ("", "x86_64-unknown-linux-gnu"),
 ];
 

